resources/environments/resource_id_broker_mappings.go

```diff
@@ -209,6 +225,16 @@ func (r *idBrokerMappingsResource) Read(ctx context.Context, req resource.ReadRe
 	}
 
 	client := r.client.Environments
+
+	if !isEnvironmentStillExists(ctx, client, state.EnvironmentName.ValueString(), &state) {
+		resp.Diagnostics.AddWarning("Resource not found on provider", "Environment not found, removing ID Broker mapping from state.")
+		tflog.Warn(ctx, "Environment not found, removing ID Broker mapping from state", map[string]interface{}{
+			"id": state.ID.ValueString(),
+		})
+		resp.State.RemoveResource(ctx)
+		return
+	}
```

Comment:
would it be possible to extract the majority of these into a separate function to decrease the amount of repetitions?

Reply:
otherwise lgtm

Reply:
+1